[DistractedCat] iP

DistractedCat opened the pull request:
# Mimi task organizer bot
Mimi frees your mind of having to remember things you need to do. It's,
- text-based
- **easy to learn**
- ~~fast~~ SUPER FAST to use

All you need is to 
1. download it from [here](https://github.com/DistractedCat/ip/releases)
2. add your tasks
3. Let Mimi handle the rest 😄 

And it's **FREE**!

Features:

- [x]  managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

DistractedCat committed:
"Level 0"

DistractedCat committed:
"Level 1"

DistractedCat committed:
"Level 2"

DistractedCat committed:
"Level 3
"

DistractedCat committed:
"Level 4"

DistractedCat committed:
"Level 5"

DistractedCat committed:
"Level 6"

DistractedCat committed:
"Level-TextUITesting"

DistractedCat committed:
"Add ability to store data into hard drive"

DistractedCat committed:
"Change time storage format to LocalDateTime and change date display format"

DistractedCat committed:
"Refactor code to be more OOP-like"

DistractedCat committed:
Merge remote-tracking branch 'origin/add-gradle-support'

DistractedCat committed:
"Add gradle support and Junit tests"

DistractedCat committed:
"Add support for find command and write java docs"

DistractedCat committed:
"Add automated style checker."

DistractedCat committed:
"Add GUI support"

DistractedCat committed:
Merge branch 'Level-10'

waseemingly commented on src/main/java/com/mimi/commands/Command.java:
```diff
@@ -0,0 +1,27 @@
+     * Checks if the command given is the command to exit.
+     * @return returns true if the command is the exit command and false otherwise
+     */
+    public boolean isExit() {
```
I liked how you used the is prefix for naming the boolean method.

DistractedCat replied:
Thank you for your encouragement!

waseemingly commented on src/main/java/com/mimi/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,67 @@
+package com.mimi.commands;
```
I liked how you used small letters to represent packages.

DistractedCat replied:
Thank you for your encouragement!

waseemingly commented on src/main/java/com/mimi/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,67 @@
+ */
+public class DeadlineCommand extends Command {
+
+    private boolean isCompleteCommand;
```
I like how you used camelCase to name your variables.

DistractedCat replied:
Thank you for your encouragement!

waseemingly commented on src/main/java/com/mimi/commands/ExitCommand.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Representation of the Exit Command.
+ */
+public class ExitCommand extends Command {
```
I like how your class case is named in PascalCase.

DistractedCat replied:
Thank you for your encouragement!

waseemingly reviewed (commented):
Overall, the code adheres to the coding standard and I am not able to notice any coding standard violations.

DistractedCat committed:
"Add java assertions"

DistractedCat committed:
Improve Code Quality.

Follows the code quality guide such as by using whitespace to separate
code logic. In switch statements, the default case is added.

DistractedCat committed:
Merge pull request #1 from DistractedCat/A-CodeQuality

Improve Code Quality.

DistractedCat committed:
Merge pull request #2 from DistractedCat/A-assertions

"Add java assertions"

DistractedCat committed:
Add support for reminders.

List of changes implemented:
1. User can no longer add tasks that are already overdue.
2. On start up, the user will automatically get a reminder on tasks that
 are due in less than a week, and also tasks previously added that has
 become overdue.
 3. The user can call for a reminder by using the remind command.

DistractedCat committed:
Add support for reminders.

List of changes implemented:
1. User can no longer add tasks that are already overdue.
2. On start up, the user will automatically get a reminder on tasks that
 are due in less than a week, and also tasks previously added that has
 become overdue.
 3. The user can call for a reminder by using the remind command.

DistractedCat committed:
Merge branch 'B-Reminders'

DistractedCat committed:
Perform small bug fixes and UI improvements

DistractedCat committed:
Merge branch 'A-betterGUI'

DistractedCat committed:
Perform small bug fixes and UI improvements

DistractedCat committed:
Make ui responses more consistent

DistractedCat committed:
Add more error handling

Users can no longer add events where the start time is before the end
time. Users can also no longer add tasks that are completely identical
to that of a previously created task.

DistractedCat committed:
Fix exit command

Programme will close after 3 seconds delay.

DistractedCat committed:
Fix exit command

Programme will close after 3 seconds delay.

DistractedCat committed:
Add files via upload

DistractedCat committed:
Update README.md

DistractedCat committed:
Update README.md

DistractedCat committed:
Fix nits and improve Code Quality

DistractedCat committed:
Merge remote-tracking branch 'origin/master'

DistractedCat committed:
Fix nits and improve Code Quality

DistractedCat committed:
Merge remote-tracking branch 'origin/master'

DistractedCat committed:
Update gradle file

DistractedCat committed:
Update README.md

DistractedCat committed:
Update README.md
